language-support/ts/codegen/tests/src/DA/Test/Daml2Ts.hs

```diff
@@ -278,5 +256,14 @@ tests damlTypes yarn damlc daml2ts davl = testGroup "daml2ts tests"
         ["  - " ++ dependency | dependency <- dependencies] ++
         ["build-options: [--target=" <> LF.renderVersion ver <> "]" | Just ver <- [mbLfVersion]]
 
+    writePackageJson :: IO ()
+    writePackageJson = BSL.writeFile "package.json" $ encode packageJson
+      where
+        packageJson = object
+          [ "private" .= True
+          , "workspaces" .= (["daml2ts"] :: [T.Text])
+          , "resolutions" .= HMS.fromList ([("@daml/types", "file:daml-types")] :: [(T.Text, T.Text)])
+          ]
```

Comment:
We have the functionality of setting up the yarn environment for tests twice now. Here and way further up. We should factor that out, probably in a followup PR.

Reply:
Well yes, we have two occasions to write json:
(1)
```haskell
        BSL.writeFile "package.json" $ encode (
          object
            [ "private" .= True
            , "devDependencies" .= pkgDevDependencies configConsts
            , "workspaces" .= pkgs
            , "name" .= ("daml2ts" :: T.Text)
            , "version" .= ("0.0.0" :: T.Text)
            ])
```
(2)
```haskell
    writePackageJson :: IO ()
    writePackageJson = BSL.writeFile "package.json" $ encode packageJson
      where
        packageJson = object
          [ "private" .= True
          , "workspaces" .= (["daml2ts"] :: [T.Text])
          , "resolutions" .= HMS.fromList ([("@daml/types", "file:daml-types")] :: [(T.Text, T.Text)])
          ]
```
Although similar, they do have very different contents. Do you still feel this deserves refactoring?

Reply:
Oh wait! I think you mean we do something similar in the DAML assistant too. Yes, we should definitely factor this out then. I'll put it on the stack with the 2 notes below.